solutions/cf-585D.mdx

```diff
@@ -0,0 +1,108 @@
+---
+id: cf-585D
+source: CF
+title: "Lizard Era: Beginning"
+author: Albert Ye
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/20898)
+
+In sum, we notice that the size of the array is 25. A naive brute-force would take $\mathcal O(3^n)$ which is too slow. But, a meet-in-the-middle solution would take at most $\mathcal O(\sqrt{3^n} \log (\sqrt{3^n}))$, which is a significant optimization. 
+
+## Implementation
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+const int MAXN = 25;
+
+int n, mid;
+// the added / removed approval from each hero for each action
+int d[MAXN][3];
+map<pair<int, int>, pair<int, int> > m;
+
+// each recursion has the index we're at, the approvals we have per hero, and the steps we took to get here
+void rec1(int i, int d0, int d1, int d2, int opt)
+{
+	if(i >= mid)
+	{
+		// find the largest happiness you can get for the first half
+		if(m.count({d0-d1, d0-d2}))
+		{
+			if(m[{d0-d1, d0-d2}].first < d0) m[{d0-d1, d0-d2}] = {d0, opt};
+		} else 
+		{
+			m[{d0-d1, d0-d2}] = {d0, opt};
+		}
+		return;
+	}
```

Comment:
recommended to keep braces on same line:
```
for () {

}
```

Reply:
oh will fix this part